core/js/src/main/scala/zio/ZIOAppPlatformSpecific.scala

```diff
@@ -10,13 +10,17 @@ trait ZIOAppPlatformSpecific { self: ZIOApp =>
    */
   final def main(args0: Array[String]): Unit = {
     implicit val trace = Tracer.newTrace
-    val newRuntime     = runtime.mapRuntimeConfig(hook)
-    newRuntime.unsafeRunAsync {
-      invokeWith(newRuntime)(Chunk.fromIterable(args0))
-        .provideEnvironment(runtime.environment)
-        .tapErrorCause(ZIO.logErrorCause(_))
-        .exitCode
-        .tap(exit)
+
+    val newLayer =
+      Scope.default +!+ ZLayer.succeed(ZIOAppArgs(Chunk.fromIterable(args0))) >>>
```

Comment:
What did we decide about replacing `+++` with `+!+`?

Reply:
I think we can replace all of these usages now that we have addressed some of the issues with tags though probably better to do that in a separate PR.